lib/server.js

```diff
@@ -108,19 +108,25 @@ exports.create = function create(createOptions) {
 
     try {
       if (this.useRouter) {
-        match(thing, function (error, redirectLocation, renderProps) {
+        match({ routes:createOptions.routes, location:thing},
+          function(error, redirectLocation, renderProps) {
           if (redirectLocation) {
+            debug('server.js match 302 %s', redirectLocation);
```

Comment:
@samsel thanks I will propose some changes. actually i already go this working, just writing some test cases. will chat more on the other pull request.